server/models/Interface.js

```diff
@@ -1,26 +1,44 @@
 const express = require("express");
 const { getNewestRedditPosts } = require("./../routes/reddit");
+const { getNewTweets } = require("./../routes/twitter");
+const Mention = require("./Mention");
 
 const mentionsInterface = class Interface {
   async getNewestMentions(companies) {
     let mentions = {
-      Reddit: []
+      Reddit: [],
+      Twitter: []
     };
 
     let promises = [];
 
     for (const company of companies) {
       promises.push(getNewestRedditPosts(company));
+      promises.push(getNewTweets(company));
     }
 
     let posts = await Promise.all(promises);
-
     for (const post of posts) {
       mentions[post[0].platform] = [...mentions[post[0].platform], ...post];
     }
-
     return mentions;
   }
+
+  async getAllMentions(companies) {
+    let allMentions = {
+      Reddit: [],
+      Twitter: []
+    };
+    for (const company of companies) {
+      await Mention.find({ company: company }, (err, mentions) => {
+        for(const mention of mentions){
+          allMentions[mention.platform].push(mention);
+        }
+      });
+    }
```

Comment:
Might want to find better way other than nested for loop, but speed seems to be quite quick still for now